Exercise_1.java

```diff
@@ -1,44 +1,74 @@
 class Stack { 
-    static final int MAX = 1000; 
+    final int MAX = 1000;
     int top; 
     int a[] = new int[MAX]; // Maximum size of Stack 
   
     boolean isEmpty() 
-    { 
-        //Write your code here 
+    {
+        if(top==-1)
+             return true;
+        else
+            return false;
     } 
 
     Stack() 
-    { 
-        //Initialize your constructor 
-    } 
+    {
+        top = -1;
+    }
   
     boolean push(int x) 
     { 
-        //Check for stack Overflow
-        //Write your code here
+        if(top<MAX-1){
+            top++;
+            a[top]=x;
+            return true;
+        }else{
+            return false;
+        }
+
     } 
   
     int pop() 
     { 
         //If empty return 0 and print " Stack Underflow"
         //Write your code here
+        if(top!=-1 && top<MAX-1){
+            int popInt = a[top];
+            a[top]=0;
+            top--;
+            return popInt;
+        }else {
+            System.out.println("Stack Underflow");
+            return 0;
+        }
+
     } 
   
     int peek() 
     { 
-        //Write your code here
-    } 
+        return a[top];

```

Comment:
what If top is -1? Please add the edge case to guard operating on an empty stack.